scalding-db/src/main/scala/com/twitter/scalding/db/TypedJdbcSource.scala

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2015 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package com.twitter.scalding.db
+
+import java.io.IOException
+import java.sql.{ ResultSet, ResultSetMetaData }
+import scala.collection.JavaConverters._
+import scala.util.{ Failure, Success, Try }
+
+import cascading.flow.FlowProcess
+import cascading.jdbc.JDBCTap
+import cascading.pipe.Pipe
+import cascading.scheme.Scheme
+import cascading.scheme.hadoop.{ TextLine => CHTextLine }
+import cascading.scheme.local.{ TextLine => CLTextLine }
+import cascading.tap.Tap
+import cascading.tap.hadoop.Hfs
+import cascading.tuple.{ Fields, Tuple, TupleEntry, TupleEntryCollector, TupleEntryIterator }
+import cascading.util.Util
+
+import org.apache.hadoop.fs.Path
+import org.apache.hadoop.mapred.{ JobConf, OutputCollector, RecordReader }
+
+import com.twitter.bijection.Injection
+import com.twitter.scalding._
+
+/**
+ * Extend this source to let scalding read from or write to a database.
+ * In order for this to work you need to specify the table name, column definitions and DB credentials.
+ * If you write to a DB, the fields in the final pipe have to correspond to the column names in the DB table.
+ *
+ * NOTE: Currently, only MySQL is supported.
+ *
+ * Example usage:
+ * case object YourTableSource extends JDBCSource {
+ *   override val tableName = TableName("tableName")
+ *   override val columns = List(
+ *      varchar("col1", 64),
+ *      date("col2"),
+ *      tinyint("col3"),
+ *      double("col4")
+ *   )
+ *   override def currentConfig = ConnectionConfig(
+ *     ConnectUrl("jdbc:mysql://mysql01.company.com:3306/production"),
+ *     MysqlDriver
+ *   )
+ * }
+ *
+ * @author Ian O Connell
+ * @author Ruban Monu
+ */
+
+abstract class TypedJDBCSource[T <: AnyRef: DBTypeDescriptor: Manifest](dbsInEnv: AvailableDatabases)
```

Comment:
would be awesome if we didn't need Manifest here (especially given that we have the DBTypeDescriptor). Manifest is deprecated and is going to cause us pain sooner or later.

Also, do we really need the `T <: AnyRef`? Ideally just the TypeDescriptor would do.